### ISSUE TITLE
[Feature] Speaker verification

### ISSUE THREAD

**Issue by urroxyz** [facebookresearch/omnilingual-asr#23]
### Feature Description

I've found that the pre-trained Wav2Vec models can be used for speaker verification even without additional fine-tuning. 

Initial experiments showed high inter-speaker similarity (around 0.95), proving the base features are content-centric, which is expected for ASR models, and usually impossible to circumvent with systems like OpenAI's Whisper.

However, I overcame this by implementing a statistical pooling of the embeddings (*μ* + *σ*), pre-processing (i.e., volume normalization), and symmetric score normalization (s-norm) against a small and diverse dataset.

This approach successfully created newly disentangled speaker verification results, yielding clear separation: same-speaker scores moved consistently above a 2.5 normalized score, while different-speaker scores were correctly suppressed below 1.5.

I write this issue to ask if a PR implementing this feature—perhaps as an extension to the `.models.inference` module or its own import (`.models.verification`, `.models.embeddings`)—would be something the maintainers would consider for the library?

If yes, I can work on it as soon as possible!

### Use Case

Speaker verification is namely associated with diarization, or the question of "who said what and when?"

The feature is highly requested in popular ASR libraries because it is an important piece of information when dealing with automatic transcription.
